tools/build_script_generator/cmake/resources/ModmConfiguration.cmake.in

```diff
@@ -0,0 +1,160 @@
+    $<$<COMPILE_LANGUAGE:CXX>:$<$<CONFIG:Release>:${ARCHFLAGS_RELEASE} ${CCFLAGS_RELEASE} ${CXXFLAGS} ${CXXFLAGS_RELEASE}>>
+    $<$<COMPILE_LANGUAGE:CXX>:$<$<CONFIG:Debug>:${ARCHFLAGS_DEBUG} ${CCFLAGS_DEBUG} ${CXXFLAGS} ${CXXFLAGS_DEBUG}>>
+
+    $<$<COMPILE_LANGUAGE:C>:$<$<CONFIG:MinSizeRel>:${ARCHFLAGS_RELEASE} ${CCFLAGS_RELEASE} ${CFLAGS} ${CFLAGS_RELEASE}>>
+    $<$<COMPILE_LANGUAGE:C>:$<$<CONFIG:Release>:${ARCHFLAGS_RELEASE} ${CCFLAGS_RELEASE} ${CFLAGS} ${CFLAGS_RELEASE}>>
+    $<$<COMPILE_LANGUAGE:C>:$<$<CONFIG:Debug>:${ARCHFLAGS_DEBUG} ${CCFLAGS_DEBUG} ${CFLAGS} ${CFLAGS_DEBUG}>>
+
+    $<$<COMPILE_LANGUAGE:ASM>:$<$<CONFIG:MinSizeRel>:${ARCHFLAGS_RELEASE} ${CCFLAGS_RELEASE} ${ASMFLAGS} ${ASMFLAGS_RELEASE}>>
+    $<$<COMPILE_LANGUAGE:ASM>:$<$<CONFIG:Release>:${ARCHFLAGS_RELEASE} ${CCFLAGS_RELEASE} ${ASMFLAGS} ${ASMFLAGS_RELEASE}>>
+    $<$<COMPILE_LANGUAGE:ASM>:$<$<CONFIG:Debug>:${ARCHFLAGS_DEBUG} ${CCFLAGS_DEBUG} ${ASMFLAGS} ${ASMFLAGS_DEBUG}>>)
+
+  target_compile_options(${target_warnings} INTERFACE
+    ${CCWARN}
+    $<$<COMPILE_LANGUAGE:CXX>:$<$<CONFIG:MinSizeRel>:${CCWARN_RELEASE} ${CXXWARN} ${CXXWARN_RELEASE}>>
+    $<$<COMPILE_LANGUAGE:CXX>:$<$<CONFIG:Release>:${CCWARN_RELEASE} ${CXXWARN} ${CXXWARN_RELEASE}>>
+    $<$<COMPILE_LANGUAGE:CXX>:$<$<CONFIG:Debug>:${CCWARN_DEBUG} ${CXXWARN} ${CXXWARN_DEBUG}>>
+
+    $<$<COMPILE_LANGUAGE:C>:$<$<CONFIG:MinSizeRel>:${CCWARN_RELEASE} ${CWARN} ${CWARN_RELEASE}>>
+    $<$<COMPILE_LANGUAGE:C>:$<$<CONFIG:Release>:${CCWARN_RELEASE} ${CWARN} ${CWARN_RELEASE}>>
+    $<$<COMPILE_LANGUAGE:C>:$<$<CONFIG:Debug>:${CCWARN_DEBUG} ${CWARN} ${CWARN_DEBUG}>>)
+
+  target_link_options(${target_options} INTERFACE
+    ${LINKFLAGS} ${ARCHFLAGS}
+    $<$<CONFIG:MinSizeRel>:${LINKFLAGS_RELEASE} ${ARCHFLAGS_RELEASE}>
+    $<$<CONFIG:Release>:${LINKFLAGS_RELEASE} ${ARCHFLAGS_RELEASE}>
+    $<$<CONFIG:Debug>:${LINKFLAGS_DEBUG} ${ARCHFLAGS_DEBUG}>)
+endfunction()
+
+function(write_target_files project_name)
+  if(NOT MODM_BMP_PORT)
+    set(MODM_BMP_PORT
+      auto
+      CACHE STRING "Set bmp debug port." FORCE)
+  endif()
+
+  if(NOT MODM_ITM_FCPU)
+    set(MODM_ITM_FCPU
+      0
+      CACHE STRING "Set itm fcpu." FORCE)
+  endif()
+
+  if(NOT MODM_DBG_UI)
+    message(STATUS "Setting debug ui to 'tui' as none was specified.")
+    set(MODM_DBG_UI
+        tui
+        CACHE STRING "Choose debug ui type." FORCE)
+    set_property(
+      CACHE MODM_DBG_UI
+      PROPERTY STRINGS
+              "tui"
+              "gdbgui")
+  endif()
+
+  set_target_properties(${project_name}
+    PROPERTIES SUFFIX ".elf")
+
+  add_custom_command(TARGET ${project_name}
+    POST_BUILD
+    COMMAND ${CMAKE_OBJCOPY} -Obinary ${project_name}.elf ${project_name}.bin)
+
+  add_custom_command(TARGET ${project_name}
+    POST_BUILD
+    COMMAND ${CMAKE_OBJCOPY} -Oihex ${project_name}.elf ${project_name}.hex)
+
+  add_custom_command(TARGET ${project_name}
+    POST_BUILD
+    COMMAND ${CMAKE_SIZE_UTIL} ${project_name}.elf)
```

Comment:
Why not use the `size` implementation from modm_tools here?

`${TOOLCHAIN}-size` is buggy for a lot of targets and was hence replaced by the `modm_tools/size.py` in scons.